073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,250 @@
+The proposed changes make it easier to reason about the current state and track the roll out of trust.
+
+## Proposal
+
+Strimzi should be updated to decouple the management of the cluster and clients CA certificates, and the consumption of those CA certificates by components (such as ZooKeeper, Kafka and cluster operator).
+
+To do this we will introduce new shared Kubernetes Secrets that are used to keep track of CA certificates that are currently in use and their current state with respect to issued end entity certificates.
+These new Kubernetes Secrets are separate from the ones that are used by Strimzi to store the latest CA certificate and key (or by the user when they are managing their own certificate).
+The cluster operator will update these new Kubernetes Secrets when it renews cluster or client CA certificates (or observes that the user has updated them).
+The cluster operator and user operator will copy the CA certificates from the existing Kubernetes Secrets into the new shared Kubernetes Secrets.
+The new shared Kubernetes Secrets will be mounted into components to use as a truststore.
+The cluster operator and user operator will update the new shared Kubernetes Secrets to reflect whether the CA certificates are currently trusted and/or in use or not.
+This will make it easier to reason about which CA certificates are in use/trusted, and allow the management of CA certificates to be evolved further in future (see [Future changes](#future-changes)).
+These new Kubernetes Secrets can be updated by multiple processes (i.e the cluster operator when issuing the CA certificates, and the cluster and user operators when updating the state of CA certificates).
+
+### New Kubernetes Secret format for CA certificate state
+
+A new Kubernetes Secret is introduced which is primarily updated by the cluster operator in its role as a CA, but also updated by the cluster operator and user operator to reflect whether the individual CA certificates are trusted and/or in use in the cluster.
+
+```yaml
+kind: Secret
+metdata:
+  name: ${cluster}-${ca-type}-trust # e.g. foo-cluster-ca-trust
+type: strimzi.io/trust
+data:
+  ${fingerprint}.${state}: <PEM encoded CA certificate>
+```
+
+The `fingerprint` is the fingerprint/thumbprint of the certificate. I.e. the SHA-1 hash of the ASN.1 DER-encoded form of the X509 certificate, which is a commonly used way of identifying certificates using common tooling.
+The `state` can be any of the states of the CA certificate trust state machine described below.
+Including the state in the item name facilitates understanding the current role of the Kubernetes Secret within the cluster.
+
+Problems from multiple processes concurrently updating the Kubernetes Secret are avoided by using `metadata.resourceVersion` to make conditional requests when updating that Secret. Note also that only one operator makes each state transition.
```

Comment:
How do you avoid such problems? By _loops_ of unsuccessful updates? By reporting the failure?